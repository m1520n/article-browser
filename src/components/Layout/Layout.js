import React from 'react';
import PropTypes from 'prop-types';

import { LayoutWrapper } from './Layout.style';

const propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
};

const Layout = ({ children }) => <LayoutWrapper>{children}</LayoutWrapper>;

Layout.propTypes = propTypes;

export default Layout;
