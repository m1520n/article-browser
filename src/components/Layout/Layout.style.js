import styled from 'styled-components';

import { colors } from '../../styles/variables';

export const LayoutWrapper = styled.main`
  background-color: ${colors.white};
  padding: 4.8rem 5.3rem;
`;

export default {
  LayoutWrapper,
};
