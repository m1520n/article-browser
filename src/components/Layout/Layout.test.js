import React from 'react';
import renderer from 'react-test-renderer';

import Layout from './Layout';

test('Layout is rendered correctly', () => {
  const component = renderer.create(
    <Layout>
      <div>
        <h1>TEST</h1>
      </div>
    </Layout>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
