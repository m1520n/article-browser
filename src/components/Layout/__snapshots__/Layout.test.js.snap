// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout is rendered correctly 1`] = `
<main
  className="sc-bdVaJa dEBSQn"
>
  <div>
    <h1>
      TEST
    </h1>
  </div>
</main>
`;
